app/javascript/flavours/glitch/components/name_list.js

```diff
@@ -0,0 +1,126 @@
+        href={href}
+        className='status__display-name'
+        onClick={this.handleClick}
+        dangerouslySetInnerHTML={{ __html: children }}
+      />
+    );
+  }
+
+}
+
+/**
+ * Displays a list of accounts as a comma-separated, link-ified list of displaynames.
+ */
+@injectIntl
+export default class NameList extends React.PureComponent {
+
+  static propTypes = {
+    intl: PropTypes.object,
+    accounts: ImmutablePropTypes.listOf(ImmutablePropTypes.map),
+    viewMoreHref: PropTypes.string,
+    onAccountClick: PropTypes.func,
+    onViewMoreClick: PropTypes.func,
+  };
+
+  render() {
+    const { accounts, intl, viewMoreHref, onAccountClick, onViewMoreClick } = this.props;
+
+    // render a single name if there is only one account
+    if (accounts.size === 1) {
+      return (
+        <span>
+          <NameLink
+            href={accounts.get(0).get('url')}
+            account={accounts.get(0)}
+            onClick={onAccountClick}
+          >
+            {accounts.get(0).get('display_name_html') || accounts.get(0).get('username')}
+          </NameLink>
+        </span>
+      );
+    }
+
+    // turn a list of accounts into a list (max length 3) of labels
+    let accountsToIntl;
+    if (accounts.size > 3) {
+      accountsToIntl = accounts.slice(0, 2).map(acct => acct.get('display_name_html') || acct.get('username'));
+      accountsToIntl = accountsToIntl.push(intl.formatMessage({ id: 'notifications.others', defaultMessage: 'others' }));
+    } else {
+      accountsToIntl = accounts.map(acct => acct.get('display_name_html') || acct.get('username'));
+    }
+
+    // turn the list of labels into an array of parts, with the correct localization
+    // see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat
+    const parts = new Intl.ListFormat(intl.locale, { type: 'conjunction' }).formatToParts(accountsToIntl);
+
+    // linkify the list of labels
+    let elementNum = 0;
+    return parts.map(({ type, value }) => {
+      const currentElement = elementNum;
+
+      // if this is a label, linkify it
+      if (type === 'element') {
+        elementNum++;
+
+        // special case for the "and others" label
+        if (currentElement === 2) {
+          return (
+            <NameLink
+              key={accounts.get(currentElement).get('id')}
+              href={viewMoreHref}
+              onClick={onViewMoreClick}
+            >
+              {value}
+            </NameLink>
+          );
+        }
+
+        const account = accounts.get(currentElement);
+        let href = account.get('url');
+
+        // return the linkified label
+        return <NameLink href={href} account={account} key={account.get('id')} onClick={onAccountClick}>{value}</NameLink>;
+      } else {
+        // if this is a separator, just print it out regularly
+        return <React.Fragment key={`${accounts.get(currentElement).get('id')}_separator`}>{value}</React.Fragment>;
+      }
```

Comment:
The variable `href` isn't necessary and only used once.
`account` could be defined earlier, after `const currentElement = elementNum;` and used in the others NameLink and else statement.

```suggestion
      const currentElement = elementNum;
      const account = accounts.get(currentElement);

      // if this is a label, linkify it
      if (type === 'element') {
        elementNum++;

        // special case for the "and others" label
        if (currentElement === 2) {
          return (
            <NameLink
              key={account.get('id')}
              href={viewMoreHref}
              onClick={onViewMoreClick}
            >
              {value}
            </NameLink>
          );
        }

        // return the linkified label
        return <NameLink href={account.get('url')} account={account} key={account.get('id')} onClick={onAccountClick}>{value}</NameLink>;
      } else {
        // if this is a separator, just print it out regularly
        return <React.Fragment key={`${account.get('id')}_separator`}>{value}</React.Fragment>;
      }
```